crates/polars-io/src/cloud/credential_provider.rs

```diff
@@ -625,20 +632,8 @@ mod python_impl {
         }
     }
 
-    impl Eq for PythonCredentialProvider {}
-
-    impl PartialEq for PythonCredentialProvider {
-        fn eq(&self, other: &Self) -> bool {
-            Arc::ptr_eq(&self.0, &other.0)
```

Comment:
Removed, completely unsound as a new object could be allocated in the same place

Reply:
Don't we keep that object alive?

Reply:
Ah, you're right - I got a bit confused in my head 😅